### PR TITLE
ROC-2864: Define citizen frontend infrastructure

### PR DESCRIPTION
**Pull request by rlewan**
### JIRA link ###

https://tools.hmcts.net/jira/browse/ROC-2864

### Change description ###

Initial setup for deploying Citizen Frontend.

### Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    IDAM_S2S_AUTH = "${var.service-2-service-auth-url}"
+
+    // Payments API
+    PAY_URL = "${var.payments-api-url}"
+
+    // Fees API
+    FEES_URL = "${var.fees-api-url}"
+
+    // Draft Store API
+    DRAFT_STORE_URL = "${var.draft-store-api-url}"
```
Comment (timja-kainos) [inline main.tf]: these should come from the infra modules

File: variables.tf [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+}
+
+// Module configuration
+variable "node-uv-threadpool-size" {
+  type = "string"
+  default = "64"
+}
+
+// Our service dependencies
+variable "claim-store-api-url" {
```
Comment (timja-kainos) [inline variables.tf]: how is this being set?

File: variables.tf [hunk 3]
```diff
@@ -0,0 +1,73 @@
+// Infrastructural variables
+variable "product" {
+  type    = "string"
+  default = "cmc"
+}
+
+variable "microservice" {
+  type = "string"
+  default = "claim-store"
```
[the remark this answers is not in the document]
Comment (rlewan) [reply variables.tf]: Not 100% on it.

File: variables.tf (hunk 2 above)
Comment (rlewan) [reply variables.tf]: It's the claim store infra module output.

File: main.tf (hunk 1 above)
Comment (rlewan) [reply main.tf]: So is the URL an implied output of a module?

File: variables.tf (hunk 2 above)
Comment (timja-kainos) [reply variables.tf]: oo does this automatically get set from the other module?

File: main.tf (hunk 1 above)
Comment (timja-kainos) [reply main.tf]: its not implied the module declares an output for it

Comment (rlewan) [reply main.tf]: > i.e. https://github.com/hmcts/cmc-claim-store-infrastructure/blob/master/main.tf#L31
But then you need to define pdf service module in citizen frontend, is that right? Shouldn't we be using outputs?

File: variables.tf (hunk 2 above)
Comment (rlewan) [reply variables.tf]: I don't know 😂  It makes sense to me for it to work this way.

File: main.tf (hunk 1 above)
Comment (timja-kainos) [reply main.tf]: I think that is how you get outputs, but not sure.
it passed terraform plan doing it that way

File: variables.tf (hunk 2 above)
Comment (timja-kainos) [reply variables.tf]: I don't think it will work but just merge this PR and test it out

**Pull request merged**